common/rpc/interceptor/telemetry.go

```diff
@@ -361,21 +364,68 @@ func (ti *TelemetryInterceptor) streamMetricsHandlerLogTags(
 	), []tag.Tag{tag.Operation(overridedMethodName)}
 }
 
-func (ti *TelemetryInterceptor) HandleError(
-	req interface{},
+func (ti *TelemetryInterceptor) HandleError(req interface{},
 	metricsHandler metrics.Handler,
 	logTags []tag.Tag,
 	err error,
-) {
-
+	nsName namespace.Name) {
 	metrics.ServiceErrorWithType.With(metricsHandler).Record(1, metrics.ServiceErrorTypeTag(err))
 
 	if common.IsContextDeadlineExceededErr(err) || common.IsContextCanceledErr(err) {
 		return
 	}
 
+	statusCode := serviceerror.ToStatus(err).Code()
+	logAllErrors := nsName != "" && ti.logAllReqErrors(nsName.String())
+	if !logAllErrors && isUserCaused(statusCode) {
+		return
+	}
+
+	if statusCode == codes.Internal || statusCode == codes.Unknown || logAllErrors {
```

Comment:
I remember now, we added the error hash for those two code because we mask those two error types in `MaskInternalErrorDetailsInterceptor` and we need the hash to find the actual error message. 

The additional check on `logAllErrors` is unnecessary here I think.